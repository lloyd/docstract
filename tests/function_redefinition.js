/** It's an error to declare the same function twice */

/** @function Foo */
/** @function Foo */
