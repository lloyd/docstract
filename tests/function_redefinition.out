{
  "args": [
    "function 'Foo' redefined at line 4"
  ], 
  "exception_type": "<type 'exceptions.RuntimeError'>", 
  "stack": [
    [
      "<module>", 
      "got = ds.extractFromFile(os.path.join(testDir, test + \".js\"))"
    ], 
    [
      "extractFromFile", 
      "data = self.extract(contents)"
    ], 
    [
      "extract", 
      "self._analyzeBlock(block, context, firstBlock, stack, lineStart, line)"
    ], 
    [
      "_analyzeBlock", 
      "parseData['blockHandler'].merge(doc, stack[-1][1], guessedName, self._whatContext(stack))"
    ], 
    [
      "merge", 
      "raise RuntimeError(\"function '%s' redefined\" % doc[\"name\"])"
    ]
  ]
}
